airbyte-ci/connectors/pipelines/pipelines/builds/python_connectors.py

```diff
@@ -15,13 +17,100 @@ class BuildConnectorImage(BuildConnectorImageBase):
     A spec command is run on the container to validate it was built successfully.
     """
 
+    DEFAULT_ENTRYPOINT = ["python", "/airbyte/integration_code/main.py"]
+    PATH_TO_INTEGRATION_CODE = "/airbyte/integration_code"
+
+    @property
+    def _build_connector_function(self):
+        if (
+            "connectorBuildOptions" in self.context.connector.metadata
+            and "baseImage" in self.context.connector.metadata["connectorBuildOptions"]
+        ):
+            return self._build_from_base_image
+        else:
+            return self._build_from_dockerfile
+
     async def _run(self) -> StepResult:
-        connector = await with_airbyte_python_connector(self.context, self.build_platform)
+        connector: Container = await self._build_connector_function()
+        connector = await apply_python_development_overrides(self.context, connector)
         try:
             return await self.get_step_result(connector.with_exec(["spec"]))
         except QueryError as e:
             return StepResult(self, StepStatus.FAILURE, stderr=str(e))
 
+    def _get_base_container(self) -> Container:
+        base_image_name = self.context.connector.metadata["connectorBuildOptions"]["baseImage"]
+        self.logger.info(f"Building connector from base image {base_image_name}")
+        return self.dagger_client.container(platform=self.build_platform).from_(base_image_name)
+
+    async def _provision_builder_container(self, base_container: Container) -> Container:
+        """Pre install the connector dependencies in a builder container.
+        If a python connectors depends on another local python connector, we need to mount its source in the container
+        This occurs for the source-file-secure connector for example, which depends on source-file
+
+        Args:
+            base_container (Container): The base container to use to build the connector.
+
+        Returns:
+            Container: The builder container, with installed dependencies.
+        """
+        setup_dependencies_to_mount = await find_local_python_dependencies(
+            self.context,
+            str(self.context.connector.code_directory),
+            search_dependencies_in_setup_py=True,
+            search_dependencies_in_requirements_txt=False,
+        )
+        builder = (
+            base_container.with_workdir(self.PATH_TO_INTEGRATION_CODE)
+            # This env var is used in the setup.py to know if it is run in a container or not
+            # When run in a container, we need to mount the local dependencies to ./local_dependencies
+            # The setup.py reacts to this env var and use the /local_dependencies path instead of the normal local path
+            .with_env_variable("DAGGER_BUILD", "1").with_file(
+                "setup.py", (await self.context.get_connector_dir(include="setup.py")).file("setup.py")
```

Comment:
Im not 100% positive but I think that these changes have reverted our support for pyproject.toml only connectors 

https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/actions/environments.py#L374

Reply:
We do have reverted our support but:
- No certified connector is using poetry at the momentt
- The `destination-duckdb` container will continue to be built with its Dockerfile until we set the `baseImage` metadata.

So we're not breaking anythin today.
I can add the support to poetry build in a new stack but I think it's a kind of early optimization.

Reply:
Correct me if Im wrong but because we install a python connector like a library for testing, does that not ignore the dockerfile?

Ideally I would like us to maintain the support of pyproject.toml.

Also we should be able to copy and paste that code linked relatively easy correct?